inc/api.class.php

```diff
@@ -1083,14 +1083,14 @@ protected function getItems($itemtype, $params = [], &$totalcount = 0) {
          foreach ($params['searchText']  as $filter_field => $filter_value) {
             if (!empty($filter_value)) {
                $search = Search::makeTextSearch($filter_value);
-               $where.= " AND (`$table`.`$filter_field` $search
-                               OR `$table`.`id` $search)";
+               $where.= " AND (`$table`.`$filter_field` $search)";
             }
          }
       }
 
       // filter with entity
-      if ($item->isEntityAssign()) {
+      if ($item->isEntityAssign
+              && array_key_exists('entities_id', $item->fields)) {
```

Comment:
last thing before merge for this part, i think it could be better to precise ``$item instanceof CommonDBChild`` to avoid removal in the future (seems duplicate with above line at first look) and maybe add a comment with an example (TicketTemplateMandatoryField doesn't have entities_id field)